src/Uno.UI.Runtime.Skia.Wpf/UI/Controls/UnoWpfWindowHost.cs

```diff
@@ -22,6 +22,26 @@ namespace Uno.UI.Runtime.Skia.Wpf.UI.Controls;
 internal class UnoWpfWindowHost : WpfControl, IWpfWindowHost
 {
 	private const string NativeOverlayLayerHostPart = "NativeOverlayLayerHost";
+	private static readonly Style _style = (Style)XamlReader.Parse(
+		"""
+		<Style xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+			   xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+			   xmlns:controls="clr-namespace:Uno.UI.Runtime.Skia.Wpf.UI.Controls;assembly=Uno.UI.Runtime.Skia.Wpf"
+			   TargetType="{x:Type controls:UnoWpfWindowHost}">
+			<Setter Property="Template">
+				<Setter.Value>
+					<ControlTemplate TargetType="{x:Type controls:UnoWpfWindowHost}">
+						<Border
+							Background="{x:Null}"
+							BorderBrush="{TemplateBinding BorderBrush}"
+							BorderThickness="{TemplateBinding BorderThickness}">
+							<ContentPresenter x:Name="NativeOverlayLayerHost" />
+						</Border>
+					</ControlTemplate>
+				</Setter.Value>
+			</Setter>
+		</Style>
+		""");
```

Comment:
How about convert to C# code?

```csharp
using System.Windows;
using System.Windows.Controls;
using System.Windows.Media;
using Uno.UI.Runtime.Skia.Wpf.UI.Controls;

public void CreateStyle()
{
    Style style = new Style(typeof(UnoWpfWindowHost));

    ControlTemplate template = new ControlTemplate(typeof(UnoWpfWindowHost));

    FrameworkElementFactory borderFactory = new FrameworkElementFactory(typeof(Border));

    borderFactory.SetValue(Border.BackgroundProperty, null);
    borderFactory.SetBinding(Border.BorderBrushProperty, new TemplateBindingExtension(Border.BorderBrushProperty));
    borderFactory.SetBinding(Border.BorderThicknessProperty, new TemplateBindingExtension(Border.BorderThicknessProperty));

    FrameworkElementFactory contentPresenterFactory = new FrameworkElementFactory(typeof(ContentPresenter));
    contentPresenterFactory.Name = "NativeOverlayLayerHost";

    borderFactory.AppendChild(contentPresenterFactory);

    template.VisualTree = borderFactory;

    Setter setter = new Setter(Control.TemplateProperty, template);

    style.Setters.Add(setter);
}
```

The code above generate by new bing.

Reply:
Yeah, I tried that first, but there are some templates that need to add visual state groups and I'm not really sure if that's possible with pure c#.

Reply:
Ok. Let’s first address the issues at hand, and then consider the optimization aspects.

Reply:
It may be indeed worth switching to pure C#, but we can look into it in a separate PR